src/engine/UIManager.cxx

```diff
@@ -175,9 +173,7 @@ void UIManager::closeMenu()
 
 void UIManager::addPersistentMenu(GameMenu::Ptr menu)
 {
-  auto it = std::find(m_persistentMenu.begin(), m_persistentMenu.end(), menu);
-
-  if (it == m_persistentMenu.end())
+  if (std::none_of(std::begin(m_persistentMenu), std::end(m_persistentMenu), [&](auto &&m) { return m == menu; }))
```

Comment:
Just a bit confused about the change for this line. Why the new lambda function?

Reply:
Sure, the lambda expression is needed because `std::none_of` needs a (defined) function-like expression for it to work (as a predicate).

Please know that `std::find` predicate is limited, because the predicate item it searches for, is limited to the (same) datatype as the array / vector it is scanning through.

`std::find_if` is the more encouraged generic version, which requires the same predicate requirement as `std::none_of`, a (defined) function-like expression for it to work.

(To explain what this predicate does,) the lambda expression just states that:
> for whatever is the current element `m`'s value, confirm it is the same as `menu`'s value.

This would be the same as the predicate for `std::find`.

https://en.cppreference.com/w/cpp/algorithm/find#Possible_implementation

Reply:
Awesome, thanks so much! I was kind of small brain so I appreciate you taking the time to explain!

Reply:
Ah, I'm sorry! Didn't want to make you feel small-brained at all! Just wanted to provide a complete answer to your kind question. :relaxed: 
My bad, just curious, was there a better way for next time to make the answer more co-operative (to you and others)? I enjoy working together with others in open-source projects so I'm always happy to learn different ways to share knowledge! Thanks! :smile_cat: